code/Model/SiteTree.php

```diff
@@ -3434,9 +3434,22 @@ protected function getExcludedURLSegments()
      */
     public function getAnchorsOnPage()
     {
+        $content = $this->Content ?? '';
+        
+        // Shortcodes may contain name/id attributes, they mess up the parsing below
+        // we replace them with their content first
+        // Note: We don't use ShortcodeParser::parse() here as that would execute the shortcodes
+        // which may have performance and side effects
+        $tags = ShortcodeParser::get('default')->extractTags($content);
```

Comment:
```suggestion
        $tags = ShortcodeParser::get_active()->extractTags($content);
```
A parser named "default" may not be the correct one to use.

Reply:
Thanks, I'v changed it. In the same file (SiteTree.php) in the `onBeforeWrite` it uses `ShortcodeParser::get('default')`as well. I don't know enough about the ShortcodeParsersm configs, to know if thats a problem. Just thought I'd mention it...